### DISCUSSION TITLE
v6.9.0 Insert + shouldUnregister false Bug?

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#3075] [Q&A] [answered]
Hello, I noticed what seems to be an issue. With v6.8, I was able to use insert instead of append to add a field item and then use another insert call to add more entries. But with v6.9, that does not seem to work anymore.

v6.8.6
Click append button then click duplicate button. New item/row is added.
https://codesandbox.io/s/react-hook-form-insert-at-686-s7cbc?file=/src/index.js

v6.9.0
Click append button then click duplicate button. No new items/rows are added.
https://codesandbox.io/s/react-hook-form-insert-at-690-dm0mn

v6.9.0 shouldUnregister true
Seems like it is just an issue with shouldUnregister false.
https://codesandbox.io/s/react-hook-form-insert-at-690-shouldunregister-true-icup2?file=/src/index.js

v6.9.0 but with append (what I switched to)
Click append button then click duplicate button. New item/row is added.
https://codesandbox.io/s/react-hook-form-insert-at-690-append-l4w21?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
This PR should resolve this issue: https://github.com/react-hook-form/react-hook-form/pull/3094#issuecomment-704122828

**Comment by fadi-george** [comment]
Looks like it did, thanks!
